python/lsst/analysis/tools/tasks/limitingSurfBriAnalysis.py

```diff
@@ -0,0 +1,338 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = (
+    "LimitingSurfBriTask",
+    "LimitingSurfBriConfig",
+    "LimitingSurfBriConnections",
+    "LimitingSurfBriAnalysisConnections",
+    "LimitingSurfBriAnalysisConfig",
+    "LimitingSurfBriAnalysisTask",
+)
+
+import logging
+
+import lsst.afw.math as afwMath
+import numpy as np
+from astropy.table import Table
+from lsst.pex.config import Field, ListField
+from lsst.pipe.base import PipelineTask, PipelineTaskConfig, PipelineTaskConnections, Struct
+from lsst.pipe.base.connectionTypes import Input, Output
+from lsst.skymap import BaseSkyMap
+
+from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+log = logging.getLogger(__name__)
+
+
+class LimitingSurfBriConnections(
+    PipelineTaskConnections,
+    dimensions=(),
+    defaultTemplates={
+        "detectionTableName": "",
+        "photoCalibName": "",
+        "wcsName": "",
+    },
+):
+    """Connections class for LimitingSurfBriTask."""
+
+    data = Input(
+        doc="Visit- or coadd-level object table",
+        name="{detectionTableName}",
+        storageClass="ArrowAstropy",
+        multiple=True,
+        dimensions=(),
+        deferLoad=True,
+    )
+
+    skymap = Input(
+        doc="The skymap that covers the originating data's tract.",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    photoCalib = Input(
+        name="{photoCalibName}",
+        storageClass="PhotoCalib",
+        doc="Photometric calibration associated with the originating data image.",
+        multiple=True,
+        dimensions=(),
+        deferLoad=True,
+    )
+
+    wcs = Input(
+        name="{wcsName}",
+        storageClass="Wcs",
+        doc="WCS header associated with the originating data image.",
+        multiple=True,
+        dimensions=(),
+        deferLoad=True,
+    )
+
+    limiting_surface_brightness_hist = Output(
+        name="limiting_surface_brightness_hist",
+        storageClass="ArrowNumpyDict",
+        doc="A dictionary containing the histogram values, bin mid points, and bin lower/upper edges for the "
+        "aggregated limiting surface brightness dataset, i.e. limiting surface brightnesses estimated for "
+        " all detectors in a single visit or all patches in a single tract.",
+        dimensions=(),
+    )
+
+    limiting_surface_brightness_table = Output(
+        name="limiting_surface_brightness_table",
+        storageClass="ArrowAstropy",
+        doc="A table containing two columns: the detector or patch IDs and the values of limiting surface "
+        "brightness derived for those detectors or patches.",
+        dimensions=(),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
```

Comment:
What is the below for?

Reply:
Ok, so it comes from here https://community.lsst.org/t/more-support-for-configurable-pipelinetask-connections/7630, if Jim says this is the right way to do it then I believe him.